### PR TITLE
arpack: package PARPACK

### PR DESCRIPTION
**Pull request by mmuetzel**
Package files for PARPACK in a split package to keep the dependency on MSMPI separate.

### CODE REVIEW

File: mingw-w64-arpack/PKGBUILD [hunk 1]
```diff
@@ -125,6 +131,26 @@ package_arpack() {
     sed -e "s|${MINGW_PREFIX}|\$\{_IMPORT_PREFIX\}|g" -i ${_f}
   done
 
+  if [ "${CARCH}" == "x86_64" ]; then
```
Comment (ognevny) [inline mingw-w64-arpack/PKGBUILD:134]:
```suggestion
  if [ "${CARCH}" != "i686" ]; then
```

File: mingw-w64-arpack/PKGBUILD [hunk 2]
```diff
@@ -125,6 +132,26 @@ package_arpack() {
...
+    mv ${pkgdir}${MINGW_PREFIX}/lib/pkgconfig/parpack* ${srcdir}/${MSYSTEM}-PARPACK/lib/pkgconfig/
+  fi
+}
+
+package_parpack() {
+  pkgdesc="Fortran77 subroutines designed to solve large scale eigenvalue problems with MPI (mingw-w64)"
+  depends+=("${MINGW_PACKAGE_PREFIX}-arpack"
+            "${MINGW_PACKAGE_PREFIX}-openblas"
+            "${MINGW_PACKAGE_PREFIX}-msmpi")
+
```
Comment (Biswa96) [inline mingw-w64-arpack/PKGBUILD:152]: Here should be a `provides=()` if you do not want to add provides for this sub-package.

Comment (mmuetzel) [reply mingw-w64-arpack/PKGBUILD:155]: Good point! 👍

I copied that and a couple more assignments from the `package_arpack64` from the same file.

**Comment by mmuetzel** [comment]
I downloaded and installed the build artifacts from this PR. With them, Octave's test suite still passes.
I also built ElmerFEM locally against libarpack and libparpack. It built without issues and no additional self-tests are failing afaict.

File: mingw-w64-arpack/PKGBUILD [hunk 3]
```diff
@@ -125,6 +132,29 @@ package_arpack() {
...
+    mv ${pkgdir}${MINGW_PREFIX}/lib/libparpack* ${srcdir}/${MSYSTEM}-PARPACK/lib/
+    mv ${pkgdir}${MINGW_PREFIX}/lib/pkgconfig/parpack* ${srcdir}/${MSYSTEM}-PARPACK/lib/pkgconfig/
+  fi
+}
+
+package_parpack() {
+  pkgdesc="Fortran77 subroutines designed to solve large scale eigenvalue problems with MPI (mingw-w64)"
+  depends+=("${MINGW_PACKAGE_PREFIX}-arpack"
+            "${MINGW_PACKAGE_PREFIX}-openblas"
+            "${MINGW_PACKAGE_PREFIX}-msmpi")
```
Comment (MehdiChinoune) [inline mingw-w64-arpack/PKGBUILD:151]: ${MINGW_PACKAGE_PREFIX}-msmpi is not a runtime dependency. The user should instead install the MS-MPI from Microsoft which should install msmpi.dll in `C:\Windows\system32` folder.

Comment (mmuetzel) [reply mingw-w64-arpack/PKGBUILD:151]: True. I'm not sure how to reflect that in the dependencies though.

Feel free to change that if you have an idea how to show that requirement correctly.

Comment (ognevny) [reply mingw-w64-arpack/PKGBUILD:151]: warning via .install file?

Comment (MehdiChinoune) [reply mingw-w64-arpack/PKGBUILD:151]: This affect all MPI packages, one should come up with a solution to tell users to install MS-MPI.
${MINGW_PACKAGE_PREFIX}-msmpi is a package that depends on the compilers C/C++/Fortran which is huge size to install.

**Pull request merged**